releases_page/main.go

```diff
@@ -0,0 +1,164 @@
+
+	// Create client
+	client := &http.Client{}
+	req, err := http.NewRequest("GET", githubAPIEndpoint, nil)
+	if err != nil {
+		fmt.Println("Error creating request: ", err)
+		return 1
+	}
+	// Set custom user-agent in request header
+	req.Header.Set("User-Agent", "opentofu releases page")
+	// Set token in request header
+	req.Header.Set("Authorization", "token "+token)
+
+	// Fetch releases from GitHub API
+	response, err := client.Do(req)
+	if err != nil {
+		fmt.Println("Error fetching releases: ", err)
+		return 1
+	}
+	defer response.Body.Close()
+
+	body, err := io.ReadAll(response.Body)
+	if err != nil {
+		fmt.Println("Error reading response body: ", err)
+		return 1
+	}
+
+	// Unmarshal JSON response
+	releases := []map[string]interface{}{}
+	if err := json.Unmarshal(body, &releases); err != nil {
+		fmt.Println("Error unmarshalling JSON: ", err)
+		return 1
+	}
+
+	// Create main index.html file
+	htmlFile, err := os.Create(htmlFileName)
+	if err != nil {
+		fmt.Println("Error creating main HTML file: ", err)
+		return 1
+	}
+
+	// Function to remove the first character of a string
+	funcMap := template.FuncMap{
+		"trimV": func(s string) string {
+			if len(s) <= 1 {
+				return ""
+			}
+			return s[1:]
+		},
+	}
+
+	// Parse mainPage template
+	mainPageTmpl, err := template.New("mainPageTemplate").Funcs(funcMap).Parse(mainPageTemplate)
+	if err != nil {
+		fmt.Println("Error parsing mainPage template: ", err)
+		return 1
+	}
+
+	// Execute the mainPage template and write to the main index.html
+	if err := mainPageTmpl.Execute(htmlFile, releases); err != nil {
+		fmt.Println("Error executing mainPage template: ", err)
+		return 1
+	}
+
+	// Handling error while closing main index.html file
+	if err := htmlFile.Close(); err != nil {
+		fmt.Println("Error closing main HTML file: ", err)
+		return 1
+	}
+
+	// Parse releasePage template
+	releasePageTmpl, err := template.New("releasePageTemplate").Parse(releasePageTemplate)
+	if err != nil {
+		fmt.Println("Error parsing releasePage template: ", err)
+		return 1
+	}
+
+	// Iterate over releases to create release pages
+	for _, release := range releases {
+		// Extract version from release name
+		version := release["name"].(string)[1:]
+		path := version + "/"
+
+		// Create directory for each release
+		if err := os.Mkdir(path, 0755); err != nil && !os.IsExist(err) {
+			fmt.Println(fmt.Sprintf("Error creating %s directory: ", version), err)
+			return 1
+		}
+
+		// Create index.html file for release page
+		htmlFile, err := os.Create(path + htmlFileName)
+		if err != nil {
+			fmt.Println(fmt.Sprintf("Error creating HTML file, %s%s: ", path, htmlFileName), err)
+			return 1
+		}
+
+		// Execute releasePage template and write to releases' index.html
+		if assets, ok := release["assets"].([]interface{}); ok {
+			if err := releasePageTmpl.Execute(htmlFile, assets); err != nil {
+				fmt.Println(fmt.Sprintf("Error executing releasePage template for %s%s: ", path, htmlFileName), err)
```

Comment:
This branch does not close the file properly.

Reply:
After using this file, I have closed it [here](https://github.com/opentofu/get.opentofu.org/pull/22/files#diff-04484bff761b8bd57b69beca485f38dac4512c0abe7aa0ab31615674ed3b30f1R157).

Reply:
That code will never be executed if you return with 1. This is not a problem right now because the program terminates anyway, but the code may be refactored in the future. Please make sure that you close all files on all exit paths.

Reply:
yeah got it. I'll update this.